tests/models/test_xcom_arg.py

```diff
@@ -0,0 +1,126 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from datetime import datetime, timedelta
+from unittest import mock
+
+from airflow import DAG
+from airflow.models.xcom_arg import BaseXComArg, XComArg, resolve_xcom_arg_class
+from airflow.operators.bash import BashOperator
+from airflow.operators.python import PythonOperator
+from tests.test_utils.config import conf_vars
+
+DEFAULT_ARGS = {
+    "owner": "test",
+    "depends_on_past": True,
+    "start_date": datetime(2020, 4, 22),
+    "retries": 1,
+    "retry_delay": timedelta(minutes=1),
+}
+
+VALUE = 42
+
+
+def assert_is_value(num: int):
+    assert num == VALUE
+
+
+class MockXComArg(BaseXComArg):
+    def resolve(self, context):
+        return "custom_value"
+
+
+class TestXComArg:
+    def test_xcom_pass_to_op(self):
+        with DAG(dag_id="test_xcom_pass_to_op", default_args=DEFAULT_ARGS):
+            operator = PythonOperator(
+                python_callable=lambda: VALUE,
+                task_id="return_value_1",
+                do_xcom_push=True,
+            )
+            xarg = XComArg(operator)
+            operator2 = PythonOperator(
+                python_callable=assert_is_value,
+                op_args=[xarg],
+                task_id="assert_is_value_1",
+            )
+            operator >> operator2
+        operator.run(ignore_ti_state=True, ignore_first_depends_on_past=True)
+        operator2.run(ignore_ti_state=True, ignore_first_depends_on_past=True)
```

Comment:
It's a slightly long-running test I think

Reply:
Any other good method of executing operators? `dag.run` is taking even longer :(

Reply:
Managed to shorten to 10s from 1.5m